commands/publish/index.js

```diff
@@ -144,10 +144,58 @@ class PublishCommand extends Command {
         )
       : [this.packagesToPublish];
 
+    const isBreakingChange = (currentVersion, nextVersion) => {
+      const releaseType = semver.diff(currentVersion, nextVersion);
+      switch (releaseType) {
+        case "major":
+          return true;
+        case "minor":
+          return semver.lt(currentVersion, "1.0.0");
+        case "patch":
+          return semver.lt(currentVersion, "0.1.0");
+        case "premajor":
+        case "preminor":
+        case "prepatch":
+        case "prerelease":
+          return false;
+        default: {
+          const err = { error: "not expected release type" };
+          throw err;
+        }
+      }
+    };
+
     const tasks = [
       () => this.getVersionsForUpdates(),
       versions => {
-        this.updatesVersions = versions;
+        const packages =
+          this.filteredPackages.length === this.packageGraph.size
+            ? this.packageGraph
+            : new Map(this.filteredPackages.map(({ name }) => [name, this.packageGraph.get(name)]));
+        if (this.project.isIndependent() || versions.size === packages.size || this.options.canary) {
+          // independent, force-publish=*, or canary, we don't bump all
+          this.updatesVersions = versions;
+        } else {
+          // fixed mode, note that versions are now all set to highest version;
+          let nextVersion;
```

Comment:
`this.globalVersion` already has this value, there's no need to collect it (it's already the next major, in a fixed version scenario)